types/params.go

```diff
@@ -48,7 +45,7 @@ func DefaultEvidenceParams() tmproto.EvidenceParams {
 	return tmproto.EvidenceParams{
 		MaxAgeNumBlocks: 100000, // 27.8 hrs at 1block/s
 		MaxAgeDuration:  48 * time.Hour,
-		MaxNum:          50,
+		MaxBytes:        1048576, // 1MB
```

Comment:
Why does this need to be a consensusparam? The application should never change this, right?

Reply:
It can if it wants do. Tbh I was also thinking that it might be better if this doesn't exist and therefore evidence, like txs just take up the rest of the block but just with a higher priority. In practice, unless you have a really small block size, evidence which basically never occurs anyway wouldn't be able to fill the entire block.

Reply:
The approach you outlined seems to be a better design than the current implementation. Should this be done as a follow up?

Reply:
Probably best to do it now then EvidenceParams doesn't change at all from 0.33.8 and it's not too much of a change. What do you think @melekes?

Reply:
generally I'm in favor of removing extra config variables. but what if someone DoS es Tendermint with evidence? will it be the responsibility of tendermint to restrict the number? one evidence per validator? what if evidence end up taking whole block? I guess it's fine

Reply:
Yeah I guess we do open ourselves to DOS attacks where someone could do a light client attack on their own light client to fill up the network with evidence and because it's from some validator set in the past there's not a lot we could do - but I think we would only do one evidence per validator per block in this case

Reply:
I'm also in favor of less config variables

Reply:
> generally I'm in favor of removing extra config variables. but what if someone DoS es Tendermint with evidence? will it be the responsibility of tendermint to restrict the number? one evidence per validator? what if evidence end up taking whole block? I guess it's fine

In this case would the evidence make it into the block? If it makes it into the block then it means its valid evidence, right? We can have an upper bound of allowed allocation in a block to help prevent this

Reply:
> We can have an upper bound of allowed allocation in a block to help prevent this
Yeah this is what maxBytes is :)

Reply:
Sorry, meant not as bytes but a fraction of the default block size. The application should not have control over this IMO. An alternative option is to handle it like block.TimeIotaMs, consensusparam not exposed to app.

Reply:
We would come somewhat full circle then. Originally evidence size was set to 10% block size (although all of this was allocated statically)

Side question: what does TimeIotaMs mean?

Reply:
I agree that this is something that the app shouldn't really have to worry about hence I think having no upper bound to be okay and just limiting evidence in a block as one per validator. There might be strange DOS angles but such is the nature of Tendermint